redbot/core/dev_commands.py

```diff
@@ -299,7 +299,10 @@ async def repl(self, ctx):
             else:
                 value = stdout.getvalue()
                 if result is not None:
-                    msg = "{}{}".format(value, result)
+                    try:
+                        msg = "{}{}".format(value, result)
+                    except:
```

Comment:
If this is the way this is handled (and there isn't some other part of the code that can be changed to address this issue), this should *at least* be catching the actual error, rather than a bare `except`.

Reply:
I had left this except bare because the error being raised in `__repr__()` could be any exception. This is also the place where the value from `__repr__()` is formatted so that's why the issue is being addressed here.

Reply:
It should at least be `except Exception` in that case, as to not catch `KeyboardInterrupt` and other `BaseException` exceptions and prevent the bot from exiting as expected.